x-pack/plugins/transform/public/app/sections/create_transform/components/step_define/common/get_agg_name_conflict_toast_messages.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+
+import { AggName, PivotAggsConfigDict, PivotGroupByConfigDict } from '../../../../../common';
+
+export function getAggNameConflictToastMessages(
+  aggName: AggName,
+  aggList: PivotAggsConfigDict,
+  groupByList: PivotGroupByConfigDict
+): string[] {
+  if (aggList[aggName] !== undefined) {
+    return [
+      i18n.translate('xpack.transform.stepDefineForm.aggExistsErrorMessage', {
+        defaultMessage: `An aggregation configuration with the name '{aggName}' already exists.`,
+        values: { aggName },
+      }),
+    ];
+  }
+
+  if (groupByList[aggName] !== undefined) {
+    return [
+      i18n.translate('xpack.transform.stepDefineForm.groupByExistsErrorMessage', {
+        defaultMessage: `A group by configuration with the name '{aggName}' already exists.`,
+        values: { aggName },
+      }),
+    ];
+  }
+
+  const conflicts: string[] = [];
+
+  // check the new aggName against existing aggs and groupbys
+  const aggNameSplit = aggName.split('.');
+  let aggNameCheck: string;
+  aggNameSplit.forEach(aggNamePart => {
+    aggNameCheck = aggNameCheck === undefined ? aggNamePart : `${aggNameCheck}.${aggNamePart}`;
+    if (aggList[aggNameCheck] !== undefined || groupByList[aggNameCheck] !== undefined) {
+      conflicts.push(
+        i18n.translate('xpack.transform.stepDefineForm.nestedConflictErrorMessage', {
+          defaultMessage: `Couldn't add configuration '{aggName}' because of a nesting conflict with '{aggNameCheck}'.`,
+          values: { aggName, aggNameCheck },
+        })
+      );
+    }
+  });
+
+  if (conflicts.length > 0) {
+    return conflicts;
+  }
+
+  // check all aggs against new aggName
+  Object.keys(aggList).some(aggListName => {
+    const aggListNameSplit = aggListName.split('.');
+    let aggListNameCheck: string;
+    return aggListNameSplit.some(aggListNamePart => {
+      aggListNameCheck =
+        aggListNameCheck === undefined ? aggListNamePart : `${aggListNameCheck}.${aggListNamePart}`;
+      if (aggListNameCheck === aggName) {
+        conflicts.push(
+          i18n.translate('xpack.transform.stepDefineForm.nestedAggListConflictErrorMessage', {
+            defaultMessage: `Couldn't add configuration '{aggName}' because of a nesting conflict with '{aggListName}'.`,
+            values: { aggName, aggListName },
+          })
+        );
```

Comment:
I don't think it's a good practice when `some` method contains a side effect, in this case, push to the collection. Consider using `for..of` with `break` statement

Reply:
Good find, this was some older code I copied, changed it in 9d6f488.

Reply:
just checked, please use loop label names, check https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/break for examples